### PR TITLE
do not start workers if no activities or workflows are registered

### PR DESCRIPTION
**Pull request by mantas-sidlauskas**
**What changed?**
Worker is not started if no workflows or activities are registered. We will print info message instead.

**Why?**
To reduce resource wasting 

**How did you test it?**
Unit tests

**Potential risks**

### CODE REVIEW

File: internal/internal_worker.go [hunk 1]
```diff
@@ -821,7 +839,10 @@ func (aw *aggregatedWorker) Start() error {
 		}
 	}
 
-	aw.logger.Info("Started Worker")
+	if workerStarted {
```
Comment (meiliang86) [inline internal/internal_worker.go:842]: Remove this log. Instead log after workflowWorker.Start() and activityWorker.Start() individually.

File: internal/internal_worker.go [hunk 2]
```diff
@@ -783,28 +795,34 @@ func (aw *aggregatedWorker) Start() error {
...
 	}
 
+	workerStarted := false
+
 	if !isInterfaceNil(aw.workflowWorker) {
 		if len(aw.registry.getRegisteredWorkflowTypes()) == 0 {
-			aw.logger.Warn(
-				"Starting worker without any workflows. Workflows must be registered before start.",
+			aw.logger.Info(
+				"Worker has no workflows registered. Workflows must be registered before start. Skipping",
```
Comment (meiliang86) [inline internal/internal_worker.go:803]: "Worker has no workflows registered. Workflows must be registered before start. Skipping" -> "Worker has no workflows registered, so workflow worker will not be started."
Same for activity worker.

**Review by meiliang86 — APPROVED** [review]
Looks good. Minor suggestions to the log messages.

File: internal/internal_worker.go (hunk 1 above)
Comment (mantas-sidlauskas) [reply internal/internal_worker.go]: done

File: internal/internal_worker.go (hunk 2 above)
Comment (mantas-sidlauskas) [reply internal/internal_worker.go]: thanks!

**Pull request merged**